src/miniscript/decode.rs

```diff
@@ -126,6 +126,9 @@ pub enum Terminal<Pk: MiniscriptKey, Ctx: ScriptContext> {
     Thresh(usize, Vec<Arc<Miniscript<Pk, Ctx>>>),
     /// k (<key>)* n CHECKMULTISIG
     Multi(usize, Vec<Pk>),
+    // Other
+    /// `<hash> OP_CHECKTEMPLATEVERIFY OP_DROP`
+    TxTemplate(sha256::Hash),
```

Comment:
The parse function in this file should use the lex code and match to <hash> OP_TEMPLATEVERIFY OP_DROP. Note that the parses decodes the script in backwards. It will see the tokens in the reverse order. 

If you look at the parsing code for CLTV, you can adapt this easily

Reply:
Added this logic :)